### ISSUE TITLE
Cannot bind to future nested properties

### ISSUE THREAD

**Issue by akagomez** [bitovi/canjs#1657]
```
var state = new can.Map({
	parent: null
});

var helpers = {
	childExists: function (options) {
		return this.attr('parent.child') ?
			options.fn() :
			options.inverse();
	}
}

var frag = can.stache('{{#childExists}}{{/childExists}}')(state, helpers);

// Does not cause helper to be re-evaluated
state.attr('parent', {
	child: 'foo'
}); 
```

I'm not sure if this is meant to be supported, but it sure would be awesome not to have to do `this.attr('parent') && this.attr('parent.child')` for situations like these. 

Spun off from #1651. 

/cc @kylegifford

**Comment by justinbmeyer** [comment]
This is due to where can.__reading is called within can.Map.  I tried to move it into _get, where I think it belongs, but a bunch of things broke.  I don't remember why, but that is the solution.

**Comment by akagomez** [comment]
Awesome! I'll run with that.

**Comment by justinbmeyer** [comment]
related to #1231

**Comment by justinbmeyer** [comment]
Closed by #1696.
